SymPEP-0001.md

```diff
@@ -0,0 +1,222 @@
+community to propose changes to the Python language. It also takes motivations
+from other similar processes in communities adjacent to SymPy, such as the
+[NEP](https://numpy.org/neps/nep-0000.html) process for NumPy and the
+[MEP](https://matplotlib.org/devel/MEP/index.html) process for Matplotlib.
+However, the SymPEP process differs from these processes in many ways, so
+those who may already be familiar with similar processes from other
+communities should read this document to understand how it works for SymPEPs.
+
+In particular, SymPEPs are significantly less formal than Python PEPs.
+
+## Types
+
+There are three kinds of SymPEPs:
+
+A **Standards Track** SymPEP describes a new feature or important change for
+SymPy.
+
+An **Informational** SymPEP describes a SymPy design issue, or provides general
+guidelines or information to the Python community, but does not propose a new
+feature. Informational SymPEPs do not necessarily represent a SymPy community
+consensus or recommendation, so users and implementers are free to ignore
+Informational SymPEPs or follow their advice.
+
+A **Process** SymPEP describes a process surrounding SymPy, or proposes a
+change to (or an event in) a process. Process SymPEPs are like Standards Track
+SymPEPs but apply to areas other than the SymPy library itself. They may
+propose an implementation, but not to SymPy’s codebase; they require community
+consensus. Examples include procedures, guidelines, changes to the
+decision-making process, and changes to the tools or environment used in SymPy
+development. Any meta-SymPEP is also considered a Process SymPEP.
+
+## Purpose
+
+The SymPEP process and corresponding documents serve several purposes:
+
+- To decide, as a community, whether a proposed change should be made.
+- To decide on the process by which a change will be implemented, for example,
+  whether the change should be implemented in several stages, whether
+  intermediate releases are required for things like deprecations, and so on.
+- To document the above discussions and decisions for future reference.
+- To document the motivations for a change from the perspective of end users
+  who may be affected by it. Here "end users" means both users of the SymPy
+  library, as well as developers working on parts of SymPy itself which may be
+  affected by the change.
+
+Importantly, **a SymPEP is not documentation** for the proposed change. End
+user documentation should be included with the implementation of the feature
+in the corresponding SymPy documentation. This also means that other
+documentation should not cross-reference a SymPEP as if it were documentation
+for a feature. Even technical discussion of a feature should be documented
+separately from a SymPEP. The reason is that SymPEPs will necessarily include
+details that are irrelevant to the final implementation, such as discussions
+of alternate implementations which were rejected and discussions of
+implementation of the change.
```

Comment:
I'm not sure it is clear how we know whether a particular topic needs a PEP or simply needs a PR. I guess that we should start with normal PRs and if they get bogged down , then it probably indicates that maybe a PEP is needed. Or maybe there is a way to give some guidelines on what types of things would need a PEP, for example:

- Making a SymPy 2.0
- Introducing new guidelines/policy (like the docstring style guide)
- Introducing things that cause the developers to significantly adjust their development practices (like switching to pytest).
- Introducing/removing hard dependencies.